src/qiskit_iqm/iqm_backend.py

```diff
@@ -50,17 +50,27 @@ def max_circuits(self) -> int:
         return 1
 
     def run(self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], **options) -> IQMJob:
-        if isinstance(run_input, list) and len(run_input) > 1:
-            raise ValueError('IQM backend currently does not support execution of multiple circuits at once.')
-        circuit = run_input if isinstance(run_input, QuantumCircuit) else run_input[0]
+        if self.client is None:
+            raise RuntimeError('Session to IQM client has been closed.')
+
+        circuits = [run_input] if isinstance(run_input, QuantumCircuit) else run_input
 
         qubit_mapping = options.get('qubit_mapping', self.options.qubit_mapping)
         shots = options.get('shots', self.options.shots)
 
-        circuit_serialized = serialize_circuit(circuit)
-        mapping_serialized = serialize_qubit_mapping(qubit_mapping, circuit)
+        circuits_serialized = [serialize_circuit(circuit) for circuit in circuits]
+        mappings_serialized = [
+            serialize_qubit_mapping(qubit_mapping, circuit)
+            for circuit in circuits
+        ]
+        if (
```

Comment:
What is the intended logic in this check?
Should it rather be ``any(mapping != mappings_serialized[0] ... ) or not mappings_serialized``?
When is ``mappings_serialized`` empty (or falsey)? It seems it always has the same length as ``circuits``.
Should it be `` or not mappings_serialized[0]`` instead?

Reply:
Changed it and added a test for the length of `circuits` above.